src/instruction.rs

```diff
@@ -364,6 +420,13 @@ pub struct Arithmetic {
     pub source: ArithmeticOperand,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct Logic {
+    pub operator: LogicalOperator,
+    pub operands: [MemoryReference; 2],
```

Comment:
Quil specfies `logicalBinaryOp     : ( AND | OR | IOR | XOR ) addr ( addr | INT ) ;` 

TODO: add support for `addr ( addr | INT )` operands, not `addr addr`